platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/SwirldMain.java

```diff
@@ -16,67 +16,57 @@
 
 package com.swirlds.common.system;
 
-import com.swirlds.config.api.Configuration;
-import com.swirlds.config.api.ConfigurationBuilder;
 import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.List;
 
 /**
- * To implement a swirld, create a class that implements SwirldMain. Its constructor should have no
- * parameters, and its run() method should run until the user quits the swirld.
+ * To implement a swirld, create a class that implements SwirldMain. Its constructor should have no parameters, and its
+ * run() method should run until the user quits the swirld.
  */
 public interface SwirldMain extends Runnable {
 
     /**
-     * Pass in a copy of the configuration.
+     * Get configuration types to be registered.
      *
-     * @param configuration
-     * 		configuration for this node
+     * @return a list of configuration types, ignored if null is returned
      */
-    default void setConfiguration(@NonNull final Configuration configuration) {
-        // override if needed
-    }
-
-    /**
-     * Update the configuration builder with any additional configuration that is required by this swirld.
-     *
-     * @param configurationBuilder
-     * 		the configuration builder to update
-     */
-    default void updateConfigurationBuilder(@NonNull final ConfigurationBuilder configurationBuilder) {
+    @Nullable
```

Comment:
I do not like to have a method that returns List<> to allow a nullable return. That should never be the case. Just return List.of() as default.

Reply:
change made